library/std/src/sys/windows/process.rs

```diff
@@ -527,13 +527,18 @@ impl Stdio {
             },
 
             Stdio::MakePipe => {
-                let ours_readable = stdio_id != c::STD_INPUT_HANDLE;
-                let pipes = pipe::anon_pipe(ours_readable, true)?;
+                // If stdin then make synchronous
```

Comment:
It could be useful to have a comment here explaining why we need async pipes for stdout and stderr.